packages/ui-react/src/containers/Layout/Layout.tsx

```diff
@@ -77,6 +77,17 @@ const Layout = () => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
 
+  const onLanguageChanged = useCallback((language: string) => {
+    if (document.documentElement.lang !== language) {
+      document.documentElement.lang = language;
+    }
+  }, []);
+
+  useEffect(() => {
+    i18n.on('languageChanged', onLanguageChanged);
+    return () => i18n.off('languageChanged', onLanguageChanged);
+  }, [onLanguageChanged, i18n]);
```

Comment:
Minor, you can remove the `useCallback` by placing the handler in the `useEffect`. It's not being used elsewhere, so it should be no issue.

```suggestion
  useEffect(() => {
    const onLanguageChanged = (language: string) => document.documentElement.setAttribute('lang', language);
    i18n.on('languageChanged', onLanguageChanged);
    
    return () => i18n.off('languageChanged', onLanguageChanged);
  }, [i18n]);
```

Alternatively, instead of using `on`/`off` listeners, you can react to the `i18n.language` property;

```suggestion
  useEffect(() => {
    document.documentElement.setAttribute('lang', i18n.language);
  }, [i18n.language]);
```

Reply:
I also wrote some feedback about the same thing, with another alternative approach. I think i prefer your last approach.

Reply:
Thanks for the feedback, @ChristiaanScheermeijer and @langemike. I agree that incorporating the language change directly within the `useEffect` makes for a cleaner and more straightforward implementation. The idea to react directly to the `i18n.language` property is particularly efficient and aligns well with our goal of streamlined code. I'll proceed with this approach and also adjust the `useTranslation` mock accordingly to remove the on and off functions. Appreciate your insights!